contributingGuides/STYLE.md

```diff
@@ -477,20 +477,63 @@ if (ref.current && 'getBoundingClientRect' in ref.current) {
 
 ### Default value for inexistent IDs
 
- Use `'-1'` or `-1` when there is a possibility that the ID property of an Onyx value could be `null` or `undefined`.
+Use `CONST.DEFAULT_NUMBER_ID` when there is a possibility that the number ID property of an Onyx value could be `null` or `undefined`. **Do not default string IDs to any value unless absolutely necessary**, in case it's necessary use `CONST.DEFAULT_STRING_ID` instead.
 
 ``` ts
 // BAD
-const foo = report?.reportID ?? '';
-const bar = report?.reportID ?? '0';
+const accountID = report?.ownerAccountID ?? -1;
+const accountID = report?.ownerAccountID ?? 0;
+const reportID = report?.reportID ?? '-1';
 
-report ? report.reportID : '0';
-report ? report.reportID : '';
+// BAD
+report ? report.ownerAccountID : -1;
+
+// GOOD
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const reportID = report?.reportID;
 
 // GOOD
-const foo = report?.reportID ?? '-1';
+report ? report.ownerAccountID : CONST.DEFAULT_NUMBER_ID;
+```
+
+Here are some common cases you may face when fixing your code to remove the default values.
+
+#### **Case 1**: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
+
+```diff
+-Report.getNewerActions(newestActionCurrentReport?.reportID ?? '-1', newestActionCurrentReport?.reportActionID ?? '-1');
```

Comment:
I don't get this... this would not cause any error since it is defaulting to string 😕 

I also don't get the next line since that one is not adding the `CONST.DEFAULT_NUMBER_ID`

Reply:
The new code isn't providing a default, so that causes a type error because it always expects a string, not possibly undefined.

Reply:
Yes, and we decided to not default string IDs so that's why I removed the default values